### PR TITLE
Bedtools bamtobed

### PR DESCRIPTION
**Pull request by tgaspe**
## Description

Adding new component bedtools_bamtobed.

## Issue ticket number

Closes #xxxx 

## Checklist before requesting a review

- [x] I have performed a self-review of my code

- [x] Conforms to the [Contributing guidelines](https://github.com/viash-hub/base/blob/main/CONTRIBUTING.md)

- [x] Proposed changes are described in the [CHANGELOG.md](https://github.com/viash-hub/base/blob/main/CHANGELOG.md)

- [x] I have tested my code with `viash ns test --parallel -q <name or namespace>`

- Check the correct box. Does this PR contain:
  - [ ] Breaking changes
  - [x] New functionality
  - [ ] Major changes
  - [ ] Minor changes
  - [ ] Documentation
  - [ ] Bug fixes

### CODE REVIEW

File: src/bedtools/bedtools_bamtobed/config.vsh.yaml [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        direction: output
+        description: Output BED file.
+
+  - name: Options
+    arguments:
+      - name: --bedpe
+        type: boolean_true
+        description: | 
+          Write BEDPE format.
+          - Requires BAM to be grouped or sorted by query.
```
Comment (DriesSchaumont) [inline src/bedtools/bedtools_bamtobed/config.vsh.yaml:40]:
```suggestion
          Write BEDPE format. Requires BAM to be grouped or sorted by query.
```

File: src/bedtools/bedtools_bamtobed/config.vsh.yaml [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+      - name: --input
+        alternatives: -i
+        type: file
+        description: Input BAM file.
+        required: true
+    
+  - name: Outputs
+    arguments:
+      - name: --output
+        alternatives: -o
```
Comment (DriesSchaumont) [inline src/bedtools/bedtools_bamtobed/config.vsh.yaml:29]:
```suggestion
        alternatives: -o
        required: true
```

File: src/bedtools/bedtools_bamtobed/config.vsh.yaml [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+      - name: --mate1
+        type: boolean_true
+        description: | 
+          When writing BEDPE (-bedpe) format, always report mate one as the first BEDPE "block".
+      
+      - name: --bed12
+        type: boolean_true
+        description: | 
+          Write "blocked" BED format (aka "BED12"). Forces -split.
+          http://genome-test.cse.ucsc.edu/FAQ/FAQformat#format1
```
Comment (DriesSchaumont) [inline src/bedtools/bedtools_bamtobed/config.vsh.yaml:51]:
```suggestion
          See http://genome-test.cse.ucsc.edu/FAQ/FAQformat#format1
```

File: src/bedtools/bedtools_bamtobed/config.vsh.yaml [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+            the two mapping qualities for the pair.
+          - When -ed is used with -bedpe, the total edit
+            distance from the two mates is reported.
+
+      - name: --tag
+        type: string
+        description: | 
+          Use other NUMERIC BAM alignment tag for BED score.
+          - Default for BED is to use mapping quality.
+            Disallowed with BEDPE output.
```
Comment (DriesSchaumont) [inline src/bedtools/bedtools_bamtobed/config.vsh.yaml:81]:
```suggestion
          Use other NUMERIC BAM alignment tag for BED score.
          Default for BED is to use mapping quality. Disallowed with BEDPE output.
```

File: src/bedtools/bedtools_bamtobed/script.sh [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+# Unset parameters
+[[ "$par_bedpe" == "false" ]] && unset par_bedpe
+[[ "$par_mate1" == "false" ]] && unset par_mate1
+[[ "$par_bed12" == "false" ]] && unset par_bed12
+[[ "$par_split" == "false" ]] && unset par_split
+[[ "$par_splitD" == "false" ]] && unset par_splitD
+[[ "$par_edit_distance" == "false" ]] && unset par_edit_distance
+[[ "$par_tag" == "false" ]] && unset par_tag
+[[ "$par_color" == "false" ]] && unset par_color
+[[ "$par_cigar" == "false" ]] && unset par_cigar
```
Comment (DriesSchaumont) [inline src/bedtools/bedtools_bamtobed/script.sh:15]:
```suggestion
unset_if_false=( 
  par_bedpe
  par_mate1
  par_bed12
  par_split
  par_splitD
  par_edit_distance
  par_tag
  par_color
  par_cigar
)

for par in ${unset_if_false[@]}; do
    test_val="${!par}"
    [[ "$test_val" == "false" ]] && unset $par
done
```

File: src/bedtools/bedtools_bamtobed/script.sh
Comment (DriesSchaumont) [inline src/bedtools/bedtools_bamtobed/script.sh:1]: Please make sure that the script exits on error

**Review by DriesSchaumont — CHANGES_REQUESTED** [review]

File: src/bedtools/bedtools_bamtobed/script.sh (hunk 5 above)
Comment (tgaspe) [reply src/bedtools/bedtools_bamtobed/script.sh:15]: Could I keep it as it is?

Comment (rcannood) [reply src/bedtools/bedtools_bamtobed/script.sh:17]: Or how about this?

```suggestion
unset_if_false=( 
  par_bedpe
  par_mate1
  par_bed12
  par_split
  par_splitD
  par_edit_distance
  par_tag
  par_color
  par_cigar
)

for par in ${unset_if_false[@]}; do
    [[ "${!par}" == "false" ]] && unset $par
done
```

Comment (rcannood) [reply src/bedtools/bedtools_bamtobed/script.sh:17]: @tgaspe What do you not like about @DriesSchaumont 's suggestion? :)

Comment (tgaspe) [reply src/bedtools/bedtools_bamtobed/script.sh]: It was just a stylistic choice of my part.

**Review by rcannood — APPROVED** [review]
LGTM!

**Review by rcannood — APPROVED** [review]

**Pull request merged**